src/runtime/virtcontainers/qemu_arm64.go

```diff
@@ -148,16 +159,41 @@ func (q *qemuArm64) getPFlash() ([]string, error) {
 func (q *qemuArm64) enableProtection() error {
 	q.protection, _ = availableGuestProtection()
 	if q.protection != noneProtection {
-		return fmt.Errorf("Protection %v is not supported on arm64", q.protection)
+		var err error
+		q.protection, err = availableGuestProtection()
```

Comment:
I'm struggling trying to follow the logic of this function so far.  The first line of the function calls `availableGuestProtection()` and the second line tests the result.  Since we got here, `availableGuestProtection()` didn't return `noneProtection` so it must have returned `rmeProtection` as it doesn't seem to return anything else but those two.

Now we're calling `availableGuestProtection()` again, however since it just opens `/dev/kvm` and does an `ioctl()` on it, what are the chances that it returns something different this time?  

Also, since we've got `rmeProtection` when we first called it, why not just go ahead and enable RME protection?

What am I missing?

Reply:
Hi @pmores , sorry that's my mistake here. I just remove the redundant code.